### PR TITLE
new icon: djangorest (original, plain, original-wordmark, plain-wordmark)

### LINKED ISSUE #1387: [ICON REQUEST]: Django Rest Framework
### I have searched through the issues and didn't find my problem.

- [X] Confirm

### About the icon

I think DRF can ve a very valuable icon

### Links and sources

https://www.django-rest-framework.org/

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by lunatic-fox**
## Double check these details before you open a PR

- [x] PR does not match another non-stale PR currently opened
- [x] PR name matches the format *new icon: <i>Icon name</i> (<i>versions separated by comma</i>)*. More details [here](https://github.com/devicons/devicon/blob/develop/CONTRIBUTING.md#overview)
- [x] PR's base is the `develop` branch.
- [x] Your icons are inside a folder as seen [here](https://github.com/devicons/devicon/blob/develop/CONTRIBUTING.md#organizational-guidelines)
- [x] SVG matches the standards laid out [here](https://github.com/devicons/devicon/blob/develop/CONTRIBUTING.md#svgStandards)
- [x] A new object is added in the `devicon.json` file as seen [here](https://github.com/devicons/devicon/blob/develop/CONTRIBUTING.md#-updating-the-deviconjson-)

**This PR closes #1387**

## Link to prove your SVG is correct and up-to-date.
https://www.django-rest-framework.org/img/logo.png

### CODE REVIEW

File: devicon.json [hunk 1]
```diff
@@ -1923,6 +1923,33 @@
...
+                "original-wordmark",
+                "plain-wordmark"
+            ],
+            "font": [
+                "plain",
+                "plain-wordmark"
+            ]
+        },
+        "color": "#a30000",
+        "aliases": []
```
Comment (Snailedlt) [inline devicon.json:1951]: Instead of duplicating the SVG's you can add them as aliases here. 
See this part in the wiki to learn more: https://github.com/devicons/devicon/wiki/Updating-%60devicon.json%60#aliases-and-aliasobj
```suggestion
        "aliases": [
            {
                "base": "original-wordmark",
                "alias": "original"
            },
            {
                "base": "plain-wordmark",
                "alias": "plain"
            }
        ]
```

Btw I think it should be possible to convert the plain icons into line icons if you want to add those too. Up to you if you wanna spend time on it or not though :)
I believe @kilian-paquier knows how to do it, if you need help

File: icons/djangorest/djangorest-plain.svg [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<svg viewBox="0 0 128 128" xmlns="http://www.w3.org/2000/svg">
```
Comment (Snailedlt) [inline icons/djangorest/djangorest-plain.svg:1]: Also a duplicate by the looks of it. Take a look at my comment on devicon.json

**Review by Snailedlt — CHANGES_REQUESTED** [review]

File: icons/djangorest/djangorest-original.svg [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<svg viewBox="0 0 128 128" xmlns="http://www.w3.org/2000/svg">
```
[the remark this answers is not in the document]
Comment (Snailedlt) [reply icons/djangorest/djangorest-original.svg:1]: My bad, it's not a duplicate. However I couldn't find the logo you've added as wordmark anywhere, so I still think this should be an alias for the wordmark version

File: devicon.json (hunk 1 above)
Comment (lunatic-fox) [reply devicon.json:1951]: I got your idea, but I have some proposals:

`original` - is "django" and "framework" texts, styled word "rest" centered and some sketch lines on the background.

`original-wordmark` - same as original, but without sketch lines on the background. But since they look quite the same I got this idea:

From `original` I can make a new `original-wordmark` with the word "rest" filled, looking more like a font.
bitmap

The same idea can be applied on `plain-wordmark`.

If it's ok I'll do this, otherwise I'll commit your changes and delete the similar `.svg` files.

Comment (Snailedlt) [reply devicon.json:1951]: Hmm, I while I like the wordmark version you made, I don't think we should modify their logo to be honest. Imo we should just use the original and have an alias from original to original-wordmark.

I'd like to get some more opinions on this though.
@Panquesito7 @BenSouchet What do you guys think of @lunatic-fox 's proposal?

Comment (BenSouchet) [reply devicon.json:1951]: Hi,
In my opinion the `original` is the one **with** the sketch lines, so in fine won't be in the generated devicon font due to no very lisible lines.
The `plain` should be an alias to the `plain-wordmark` and should be the logo **without** the sketch lines.
So in conclusion two different icons, the `original` and the `plain` / `plain-wordmark`.

Comment (lunatic-fox) [reply devicon.json:1951]: Seems alright to me!

Just to be extra sure:
`original` - Keep the existing one.
`plain` - The `plain` version of `original` without background sketch lines.
`plain-wordmark` - An alias of `plain` version.

Comment (Snailedlt) [reply devicon.json:1951]: > Hi, In my opinion the `original` is the one **_with_** the sketch lines, so won't be in the generated devicon font due to no very lisible lines. The `plain` should be an alias to the `plain-wordmark` and should be the logo **_without_** the sketch lines. So in conclusion two different icons, the `original` and the `plain` / `plain-wordmark`.

Agreed with the `original` icon, it should be the one **with** the sketch lines.

I'm not so sure about the plain icon though. @lunatic-fox If I understand it correctly, the one **without** sketchlines is an icon you've made yourself right? That doesn't sit right with me, since most companies don't like it when you change their logos.

### In summary I would suggest the following:
- `original`:  Keep the one **with** sketchlines and colored text
- `original-wordmark`: An alias of `original`
- `plain`: Same as `original`, but mono-colored
- `plain-wordmark`: An alias of `plain`

Comment (lunatic-fox) [reply devicon.json:1951]: I've discarded my last proposal of filling the "rest" word and stick with this idea:
bitmap1

Comment (BenSouchet) [reply devicon.json:1951]: @Snailedlt I'm not as stuck on the idea of editing a bit logo to create icon for Devicon since, for Django REST it's an logo not an icon, so for me removing the sketchlines isn't "editing" the logo, someone looking at it still recognize Django REST logo and it's fit correctly with Devicon.
Agree that `original-wordmark` could be an alias to `original` but since it won't be added to the final generated font it's doesn't matter too much.
The last proposition of @lunatic-fox is for me the best in my option / compromise.

Comment (Snailedlt) [reply devicon.json:1951]: @BenSouchet I think we just disagree then. I don't see any advantage of adding an icon which is not the official one, when we have the official one right there.

I think @Panquesito7 should also help decide though, so I'll wait for him to respond here

Comment (BenSouchet) [reply devicon.json:1951]: A **distinction** need to be made between a logo (what we have as entry point with Django REST), and a icon which must be able to be used in small (_favicon for example_), even more if the icon will be added as glyph in a font, with icons the more unnecessary details are added the less you have in readability.

Also if Django REST aren't happy by the small edit made to add the logo as icon to Devicon they can comment / open an issue / contact.

Other point, the logo isn't registered / trademarked so this isn't any illegal action at all.

Lastly, you can contact them (Django REST community) via the official Github repo asking them if its okay or not to do at small simplification, so the readability of the glyph would be better in Devicon generated font. Or do they have a simplified version that they use for icon / favicon.

Comment (kilian-paquier) [reply devicon.json:1951]: Hello, is there still a need of information for the outline part ?

Comment (Panquesito7) [reply devicon.json:1951]: > > Hi, In my opinion the `original` is the one **_with_** the sketch lines, so won't be in the generated devicon font due to no very lisible lines. The `plain` should be an alias to the `plain-wordmark` and should be the logo **_without_** the sketch lines. So in conclusion two different icons, the `original` and the `plain` / `plain-wordmark`.
> 
> @BenSouchet Thanks for the review 🙇 I agree with the `original` icon, it should be the one **with** the sketch lines.
> 
> I'm not so sure about the plain icon though. @lunatic-fox If I understand it correctly, the one **without** sketchlines is an icon you've made yourself right? That doesn't sit right with me, since most companies don't like it when you change their logos.
> 
> ### In summary I would suggest the following:
> * `original`:  Keep the one **with** sketchlines and colored text
> * `original-wordmark`: An alias of `original`
> * `plain`: Same as `original`, but mono-colored
> * `plain-wordmark`: An alias of `plain`

This sounds good to me. 👍

Comment (Snailedlt) [reply devicon.json:1951]: @kilian-paquier 
> Hello, is there still a need of information for the outline part ?

Yeah, would love to hear your opinion too :)

Comment (Snailedlt) [reply devicon.json:1951]: @kilianpaquier Please take a look when you have time :)

Comment (kilianpaquier) [reply devicon.json:1951]: > @kilianpaquier Please take a look when you have time :)

Hello, thank you for the reminder. You need both "django" and "framework" outlined ?

Comment (Snailedlt) [reply devicon.json:1951]: @kilianpaquier
I'll leave that up to you to decide. As long as it adheres to our SVG-Standards and SVG-Versions wiki pages :)

Comment (kilianpaquier) [reply devicon.json:1951]: Here you go. Tell me if it's fine ?

svgviewer-output-outline

Comment (Snailedlt) [reply devicon.json:1951]: Love it!
@lunatic-fox what do you think? Should we add it?

Comment (Snailedlt) [reply devicon.json:1951]: btw @kilianpaquier, what tools and methods do you use to make those clean line icons?

Comment (kilianpaquier) [reply devicon.json:1951]: I'll let you merge the appropriate paths and put the appropriate color

svgviewer-output (1)

Comment (lunatic-fox) [reply devicon.json:1951]: I know the original does have the sketchy lines, but in that case, since it's original version I think that's understandable.

Comment (Snailedlt) [reply devicon.json:1951]: @lunatic-fox I'm fine with that :)

Comment (Snailedlt) [reply devicon.json:1951]: > I didn't keep the weird segments and dots from the logo. But it seems to be part of the official logo, I should add that version too.
> 
> @Snailedlt Inkscape. Maybe I should add a wiki part for the particularity of outline logos ?

Yes please, you should explain it in #1306 :D

Comment (lunatic-fox) [reply devicon.json:1951]: > @lunatic-fox I'm fine with that :)

Then, I think we have a consensus now.
So the PR would be:
`original` - with all the features
`plain` - without the sketchy lines

I think it's possible to discard `original-wordmark` and `plain-wordmark`, isn't it? And they can be aliased from `plain` version.

Comment (Snailedlt) [reply devicon.json:1951]: hmmm, yesn't. What if I want the mono colored version of the original? :O I'm not sure. I think bottom line is that with only the versions we're working with in devicon we can't make everyone happy. 
So we either go with your first option:
1.  `wordmark` versions are without the sketchlines, and non-wordmark versions are with sketchlines.
2.  `original` has sketchlines, `plain` and `line` do not. Then we make aliases for `wordmark` versions.
3.  All of the non-wordmark versions have sketchlines. Then we make aliases for `wordmark` versions.

Personally leaning more towards 1. and 3. still. I kinda actually prefer 1. now that I think of it, since that leaves us with the most options, even if it's not entirely conventional

Comment (Snailedlt) [reply devicon.json:1951]: I also think 1. makes almost as much conventional sense as 3., so definitely the best option for me

Comment (lunatic-fox) [reply devicon.json:1951]: Loved it! I'll make the change based on 1.

Just to confirm it'll be like:
`original` - regular icon
`plain` - monocolored original
`original-wordmark` - `original` without sketch lines
`plain-wordmark` - monocolored `original-wordmark`

Comment (Snailedlt) [reply devicon.json:1951]: Great! Glad we can agree :D
I think it's good we had this discussion, definitely helps understand some more usecases, and intricacies of devicon

Comment (Snailedlt) [reply devicon.json:1951]: > So we will be adding @kilianpaquier line versions?
Yes! 💯 
Make sure you add him as a co-author

Comment (lunatic-fox) [reply devicon.json:1954]: All done!
I was just refining some curves, because Inkscape sometimes do weird paths.

File: devicon.json [hunk 4]
```diff
@@ -1923,6 +1923,36 @@
...
+        "tags": [
+            "framework",
+            "rest",
+            "api",
+            "python",
+            "web"
+        ],
+        "versions": {
+            "svg": [
+                "original",
```
Comment (Snailedlt) [inline devicon.json:1940]: The perfect version list doesn't exi-

File: devicon.json (hunk 1 above)
Comment (Snailedlt) [reply devicon.json:1954]: Noice!
Btw, could you add an empty line at the end for each file? It's needed for some old IDE's

**Review by Snailedlt — CHANGES_REQUESTED** [review]
Last suggestions hopefully

File: devicon.json (hunk 4 above)
Comment (lunatic-fox) [reply devicon.json:1940]: Sorry, but I don't get it.

Comment (Snailedlt) [reply devicon.json:1940]: Just very nice to see all the versions there :)
So no complaints, just wanted to make a positive comment for once ;)

File: devicon.json (hunk 1 above)
Comment (lunatic-fox) [reply devicon.json:1954]: > Noice! Btw, could you add an empty line at the end for each file? It's needed for some old IDE's

Hum, to me there's a empty line at the end of the file, maybe it's the Windows return carriage issue.
I'll be fixing that in a moment!

Comment (Snailedlt) [reply devicon.json:1954]: I meant for the icons. Maybe, if you're using linux the return carriage is different than on windows.

Not a huge issue though, since there are tons of icons already without it. I kinda wanna add an empty line to all of them eventually though, so would be nice if future PR's had that empty line :)

Comment (lunatic-fox) [reply devicon.json:1954]: Got it! I'll be doing that later! 🙂

Comment (lunatic-fox) [reply devicon.json:1954]: Done via GitHub Codespaces!
It looks like GitHub cosmetically hides the final empty line. But if you try to edit the file directly, you will see this end line.

Comment (lunatic-fox) [reply devicon.json:1954]: Done via GitHub Codespaces!
It looks like GitHub cosmetically hides the final empty line. But if you try to edit the file directly, you will see this end line.

***\* Suggestion***
&emsp;Maybe we can add this code below in `.vscode/settings.json` of #1487, then it'll automatically insert an empty line at the end of a `xml` file **when saving it (trigger)**. `xml` language was selected, because `svg` is some kind of `xml` file and `svg` it's not specified as a language in `settings.json`.
> 
```json
"[xml]": {
  "files.insertFinalNewline": true,
}
```

File: devicon.json (hunk 4 above)
Comment (lunatic-fox) [reply devicon.json:1940]: Oh, I got the mood! 😅
I just got a little bit confused because I thought about `apache` icon.

**Review by Snailedlt — DISMISSED** [review]
LGTM! ✔️ 
Great work to everyone involved! 💯 It's not every day that we get to add all icon versions for a logo, let alone in the same PR!
Special thanks to @BenSouchet @lunatic-fox @kilianpaquier and @Panquesito7 for contributions both in the form of code, reviews and discussions!

File: devicon.json (hunk 1 above)
Comment (Snailedlt) [reply devicon.json:1954]: > It looks like GitHub cosmetically hides the final empty line

Yup, though it is possible to see still, since GitHub adds a little icon when the endline is missing:
image

But it can be unreliable sometimes it seems. Notice the endline and no-endline icons in the same file here: 
image

Comment (Snailedlt) [reply devicon.json:1954]: > _*** Suggestion**_  Maybe we can add this code below in `.vscode/settings.json` of #1487, then it'll automatically insert an empty line at the end of a `xml` file **when saving it (trigger)**. `xml` language was selected, because `svg` is some kind of `xml` file and `svg` it's not specified as a language in `settings.json`.
> 
> > 
> 
> ```json
> "[xml]": {
>   "files.insertFinalNewline": true,
> }
> ```

Sounds like a good idea, just contemplating whether we want endlines added to all files, or just xml files

File: devicon.json [hunk 5]
```diff
@@ -1923,6 +1923,36 @@
...
+                "plain",
+                "line",
+                "original-wordmark",
+                "plain-wordmark",
+                "line-wordmark"
+            ],
+            "font": [
+                "plain",
```
Comment (Snailedlt) [inline devicon.json:1948]: Whoops, we forgot to add the line icon as a font :)
```suggestion
                "plain",
                "line",
```

**Review by Snailedlt — APPROVED** [review]
re-approved, hope it's good now :)
